index.js

```diff
@@ -0,0 +1,57 @@
+const express = require('express');
+const cors = require('cors');
+
+const bcrypt = require('bcryptjs');
+
+const db = require('./database/dbConfig.js')
+
+const server = express();
+
+const PORT = 5050;
+
+server.use(express.json());
+server.use(cors());
+
+
+server.post('/api/register', (req, res) =>{
+    const user = req.body;
+    const hash = bcrypt.hashSync(user.password, 14);
+    user.password = hash;
+    if(!user.username || !user.password){res.status(400).json({message:'Please provide username and password!'})}
+    db.insertUser(user)
+    .then(ids =>{
+        const id = ids[0]
+        res.json({newUserId: id})
+    })
+    .catch(err =>{
+        res.status(500).json({err})
+    })
+})
+
+server.post('/api/login', (req, res)=>{
+    const creds = req.body;
+    db.findByUsername(creds.username)
+    .then(user =>{
+        if(user && bcrypt.compareSync(creds.password, user.password)){
+            res.json({welcome:user.username})
+        }else{
+            res.status(401).json({message:"Invalid username or password!"})
+        }
+    })
+    .catch(err =>{
+        res.status(err =>{
+            res.status(500).json({err})
+        })
```

Comment:
This code block looks a little odd; why are you trying to pass a callback function to the `status` method? I'm assuming you wanted lines 41-45 to look a bit more like this:
```javascript
.catch(err => {
  res.status(500).json({ err });
});
```